homework.md

```diff
@@ -20,48 +20,84 @@ CREATE TABLE stations
 );
 ```
 
-1) Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.
+1. Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.
 
-Hint: your query will look something like 
+Hint: your query will look something like
 
-``` sql
+```sql
 INSERT INTO stations (SELECT ... FROM tripe);
 ```
 
+``` sql
+INSERT INTO stations
+(SELECT  from_station_id as id, max(distinct from_station_name)  as name
+from trips
+
+WHERE from_station_id IS NOT NULL group by from_station_id HAVING COUNT(distinct from_station_name) = 1 OR COUNT(distinct from_station_name) > 1
+
+union
+
+SELECT  to_station_id as id, max(distinct to_station_name)  as name
+from trips
+
+WHERE to_station_id IS NOT NULL group by to_station_id HAVING COUNT(distinct to_station_name) = 1 OR COUNT(distinct to_station_name) > 1
+
+ORDER BY id ASC);
```

Comment:
`max(distinct ...)` and `count(distinct ...)` may not be doing what you think they're doing - try digging into these a little more to see what I mean, it can be unintuitive. It currently looks like your solution is giving a result set that looks like what we want to insert, but the station names selected are not the most popular in aggregate. 

I suggest going through this in steps without feeling the need to create just one query to answer this question :) Feel free to reach out if you have any questions!